test/unit/power_manager_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+class PowerManagerTest < ActiveSupport::TestCase
+  test "should respond to all real actions with compute resource" do
+    vm_mock = mock('vm')
+
+    compute_resource_mock = mock('compute_resource')
+    compute_resource_mock.stubs(:find_vm_by_uuid).returns(vm_mock)
+
+    host = FactoryGirl.build(:host, :on_compute_resource)
+    host.unstub(:queue_compute)
+    host.stubs(:compute_resource).returns(compute_resource_mock)
+
+    PowerManager::SUPPORTED_ACTIONS.each do |action|
```

Comment:
... and therefore, if the loop below is calling SUPPORTED_ACTIONS, the stubs should be on the list of methods on the right hand side of the action_map.  You may want to either hardcode that list or call `host.power.send(:action_map).keys` perhaps.